### PR TITLE
send plugin load events for pulse, freewheel, and ima

### PR DESCRIPTION
**Pull request by dallen42**
PR to update ad plugins to create plugin loaded events for QoS

### CODE REVIEW

**Comment by pilievOoyala** [comment]
lgtm.

File: js/pulse.js [hunk 1]
```diff
@@ -362,11 +363,13 @@
                     log('No Pulse hostname found in plugin parameters or media metadata; will not attempt to show Pulse ads');
                     noPulseConfiguration = true;
                     this.ready = true;
+                    amc.reportPluginLoaded(Date.now() - this.initTime, this.name)
```
Comment (aeng7) [inline js/pulse.js]: The other plugins report plugin loaded after calling amc.onAdManagerReady. Should this plugin do the same for consistency?

Comment (dallen42) [reply js/pulse.js]: Fixed

**Comment by aeng7** [comment]
@dallen42 can we add some unit tests? LGTM otherwise

**Comment by dallen42** [comment]
@aeng7 unit tests are added

**Comment by aeng7** [comment]
LGTM

**Pull request merged**